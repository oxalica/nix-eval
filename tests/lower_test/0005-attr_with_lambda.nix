{ b, c }:
{
  a = 1;
  inherit b;
  inherit (c) d;
}
