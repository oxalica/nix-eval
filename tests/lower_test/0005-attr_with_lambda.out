Ok(
    Expr(
        Lambda {
            arg: ClosePattern {
                required_names: [
                    "b",
                    "c",
                ],
                optional_names: {},
            },
            body: Expr(
                LetIn {
                    exprs: [
                        Expr(
                            Select {
                                set: Debruijn(
                                    2,
                                ),
                                index: Expr(
                                    Literal(
                                        String(
                                            "b",
                                        ),
                                    ),
                                ),
                                or_default: None,
                            },
                        ),
                        Expr(
                            Select {
                                set: Debruijn(
                                    2,
                                ),
                                index: Expr(
                                    Literal(
                                        String(
                                            "c",
                                        ),
                                    ),
                                ),
                                or_default: None,
                            },
                        ),
                    ],
                    body: Expr(
                        LetIn {
                            exprs: [
                                Debruijn(
                                    1,
                                ),
                            ],
                            body: Expr(
                                AttrSet {
                                    entries: {
                                        "a": Expr(
                                            Literal(
                                                Int(
                                                    1,
                                                ),
                                            ),
                                        ),
                                        "b": Debruijn(
                                            2,
                                        ),
                                        "d": Expr(
                                            Select {
                                                set: Debruijn(
                                                    0,
                                                ),
                                                index: Expr(
                                                    Literal(
                                                        String(
                                                            "d",
                                                        ),
                                                    ),
                                                ),
                                                or_default: None,
                                            },
                                        ),
                                    },
                                    dynamics: [],
                                },
                            ),
                        },
                    ),
                },
            ),
        },
    ),
)